### PR TITLE
Send emails when access is expiring

### LINKED ISSUE #591: Send emails to users whose access is expiring soon

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #591

Rendered emails:
- [Single package expiring](https://apiary-test.robojackets.org/mailbook?selected=App%5CNotifications%5CDues%5CPackageExpirationReminder&variant=one-package&locale=en)
- [Multiple packages expiring](https://apiary-test.robojackets.org/mailbook?selected=App%5CNotifications%5CDues%5CPackageExpirationReminder&variant=two-packages&locale=en)

### CODE REVIEW

File: app/Mail/Dues/PackageExpirationReminder.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    /**
+     * Get the message content definition.
+     */
+    public function content(): Content
+    {
+        $packages = DuesPackage::whereDate('access_end', '>', Carbon::now())
+            ->whereDate('access_end', '<', Carbon::now()->addDays(7))
+            ->get();
+
+        $userCount = User::active()->accessInactive()->count();
```
Comment (evan10s) [inline app/Mail/Dues/PackageExpirationReminder.php:65]: Maybe I'm misunderstanding, but how can someone be active but access inactive? for an expiring dues package, would you not be inactive but access active?

Comment (kberzinch) [reply app/Mail/Dues/PackageExpirationReminder.php:65]: Oop good catch

File: app/Jobs/SendRemindersForExpiringAccessOverrides.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        User::whereDate('access_override_until', '>', Carbon::now())
+            ->whereDate('access_override_until', '<', Carbon::now()->addDays(2))
+            ->get()
+            ->each(static function (User $user): void {
+                SendReminders::dispatch($user);
```
Comment (evan10s) [inline app/Jobs/SendRemindersForExpiringAccessOverrides.php:41]: 🤔 there's no import for `SendReminders`, does the notification for this already exist?

Comment (kberzinch) [reply app/Jobs/SendRemindersForExpiringAccessOverrides.php:41]: It's in the same namespace, and it's using the existing transaction/payment reminders that are triggered from that job.

**Review by evan10s — DISMISSED** [review]

File: resources/views/mail/dues/packageexpirationreminder.blade.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
+The following dues {{ count($packages) === 1 ? 'package has an' : 'packages have' }} access end {{ \Illuminate\Support\Str::plural('date', count($packages)) }} within the next 7 days:
+
+@foreach($packages as $package)
+- {{ $package->name }} expires on {{ $package->access_end->format('l, F j') }}
+@endforeach
+
+{{ $userCount }} {{ \Illuminate\Support\Str::plural('member', $userCount) }} will lose access to RoboJackets systems.
+
+You may update the "Access End Date" for {{ count($packages) === 1 ? 'this package' : 'these packages' }} at {{ count($packages) === 1 ? route('nova.pages.detail', ['resource' => \App\Nova\DuesPackage::uriKey(), 'resourceId' => $packages->sole()->id]) : route('nova.pages.index', ['resource' => \App\Nova\DuesPackage::uriKey()]) }}.
```
Comment (JoshOldenburg) [inline resources/views/mail/dues/packageexpirationreminder.blade.php:9]:
```suggestion
You may update the "Access End Date" for {{ count($packages) === 1 ? 'this package' : 'these packages' }} at {{ count($packages) === 1 ? route('nova.pages.detail', ['resource' => \App\Nova\DuesPackage::uriKey(), 'resourceId' => $packages->sole()->id]) : route('nova.pages.index', ['resource' => \App\Nova\DuesPackage::uriKey()]) }}. If the dates are correct, you can ignore this email.
```

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
I think it would help to clarify that an action is not necessarily required.

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**